CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.12)
```

Comment:
Is this bump *necessary* for the new functionality or just routine or because it's what you happened to test with?

Reply:
I'm using the latest CMake version (at the time of writing of this comment, 3.24.1)

Cmake 3.1 is rather old, and I suspect there are already many things in this file relying on 3.2+ features. I just bumped it up to 3.12 because I'm adding the usage of the list `REMOVE_DUPLICATES` feature (that was added in CMake 3.12)

But I could be wrong too, I've not tested this with 3.12, maybe this number would need further bumping

Reply:
Thatks okay you you're adding usage of a feature you know what added in 3.12 then the bump makes sense. I just wanted to make sure it was *required*. Perhaps one would need an even newer version, but since you know this at *least* is the minimum then we're good. Thanks for answering.